lib/charms/nginx_ingress_integrator/v0/ingress.py

```diff
@@ -56,35 +70,50 @@
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 5
+LIBPATCH = 12
 
-logger = logging.getLogger(__name__)
+LOGGER = logging.getLogger(__name__)
 
-REQUIRED_INGRESS_RELATION_FIELDS = {
-    "service-hostname",
-    "service-name",
-    "service-port",
-}
+REQUIRED_INGRESS_RELATION_FIELDS = {"service-hostname", "service-name", "service-port"}
 
 OPTIONAL_INGRESS_RELATION_FIELDS = {
+    "additional-hostnames",
     "limit-rps",
     "limit-whitelist",
     "max-body-size",
+    "owasp-modsecurity-crs",
+    "owasp-modsecurity-custom-rules",
+    "path-routes",
     "retry-errors",
+    "rewrite-target",
+    "rewrite-enabled",
     "service-namespace",
     "session-cookie-max-age",
     "tls-secret-name",
 }
 
+RELATION_INTERFACES_MAPPINGS = {
+    "service-hostname": "host",
```

Comment:
Could we get these ordered alphabetically? As the list grows, it will be easier to check if something has already been added